tools/c7n_azure/tests/test_sqldatabase.py

```diff
@@ -67,3 +70,208 @@ def _assert_found_only_test_database(self, resources):
         db = resources[0]
 
         self.assertEqual(db.get('name'), 'cctestdb')
+
+
+class BackupRetentionPolicyFilterTest(BaseTest):
+
+    def setUp(self):
+        super(BackupRetentionPolicyFilterTest, self).setUp()
+
+        self.resouce_group = "test_sqlserver"
+        self.parent_key = "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/"
+        "test_sqlserver/providers/Microsoft.Sql/servers/cctestsqlserver"
+        self.name = "cctestdb"
+
+        manager = mock.Mock()
+        self.filter = BackupRetentionPolicyFilter('op-prop', 0, {}, manager)
+
+    def test_raises_key_error_without_resource_group(self):
+        with self.assertRaises(KeyError):
+            self.filter.process([{
+                ChildResourceQuery.parent_key: self.parent_key,
```

Comment:
nit: you can extract resource group name from parent key

Reply:
good point!